src/main/java/com/example/bookstore/repository/book/BookSpecificationBuilder.java

```diff
@@ -0,0 +1,51 @@
+package com.example.bookstore.repository.book;
+
+import com.example.bookstore.dto.BookSearchParametersDto;
+import com.example.bookstore.model.Book;
+import com.example.bookstore.repository.SpecificationBuilder;
+import com.example.bookstore.repository.SpecificationProviderManager;
+import lombok.RequiredArgsConstructor;
+import org.springframework.data.jpa.domain.Specification;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class BookSpecificationBuilder implements SpecificationBuilder<Book> {
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParameters) {
+        Specification<Book> spec = Specification.where(null);
+        if (searchParameters.title() != null && searchParameters.title().length > 0) {
```

Comment:
you are duplicating code here many times. for me it's a good candidate for a private method.